### ISSUE TITLE
The FindById and findByIdAndPartitionKey implementation don't comply with javadoc

### ISSUE THREAD

**Issue by apescione** [microsoft/spring-data-cosmosdb#451]
Version: 2.1.8

Both APIs FindById and findByIdAndPartitionKey should return Optional#empty() when a Entity hasn't been found, instead, a DocumentDBAccessException is thrown.

Below the documentation for findByIdAndPartitionKey  in com.microsoft.azure.spring.data.cosmosdb.repository.DocumentDbRepository<T, ID>

```
    /**
     * Retrieves an entity by its id.
     *
     * @param id must not be {@literal null}.
     * @param partitionKey partition key value of entity, must not be null.
     * @return the entity with the given id or {@literal Optional#empty()} if none found
     * @throws IllegalArgumentException if {@code id} is {@literal null}.
     */
    Optional<T> findById(ID id, PartitionKey partitionKey);
```

This behavior will force to catch always the exception when an entity is not found. 
DocumentDBAccessException may be thrown when other kind of errors occur.

**Comment by kushagraThapar** [comment]
Verified at our end, findById API is behaving correctly, but findById which takes partitionKey as well has this bug. Will fix it soon.

**Comment by kushagraThapar** [comment]
This has been fixed and released in v2.1.9. 

Will be fixed and released in v2.2.1.M2 in few days. .

**Comment by kushagraThapar** [comment]
This has been fixed in v2.2.1 as well. Closing this issue now.
